### PR TITLE
adds support for apiServerSource

### PR DESCRIPTION
**Pull request by invincibleJai**
**Fixes**: 
https://issues.redhat.com/browse/ODC-2660

**Analysis / Root cause**: 
User can't create ApiServerSource via web console

**Solution Description**: 
Adds support for ApiServerSource via web console

**Screen shots / Gifs for design review**: 
Screenshot 2020-03-27 at 1 41 04 PM

**Unit test coverage report**: 
Screenshot 2020-03-25 at 1 16 38 AM

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

cc @serenamarie125

### CODE REVIEW

**Comment by divyanshiGupta** [comment]
Screenshot from 2020-03-26 19-42-21


@invincibleJai I am getting this error while trying to create an `EventSource` of type `ApiServerSource`. Am I missing anything?

**Comment by invincibleJai** [comment]
@divyanshiGupta yeah it's my bad should have given proper values for `mode` we need `Resource` 

you can follow https://docs.google.com/document/d/1F4zECoLRvaUiigLaogbgYOiD_hB_w5hAtcXGY0MxVc4/edit

**Comment by invincibleJai** [comment]
have updated mode to Dropdown as currently it just takes two values and defaults to `Reference`

cc @divyanshiGupta  @sahil143

File: frontend/packages/knative-plugin/src/components/overview/EventSinkServicesOverviewList.tsx [hunk 1]
```diff
@@ -60,7 +60,7 @@ const EventSinkServicesOverviewList: React.FC<EventSinkServicesOverviewListProps
       ) : (
         <span className="text-muted">No services found for this resource.</span>
       )}
-      {pods?.length && <PodsOverview pods={pods} obj={obj} allPodsLink={linkUrl} />}
+      {pods?.length > 0 && <PodsOverview pods={pods} obj={obj} allPodsLink={linkUrl} />}
```
Comment (sahil143) [inline frontend/packages/knative-plugin/src/components/overview/EventSinkServicesOverviewList.tsx]: If pods length is 0, don't we want to show empty msg? `No pods found for this resource`

Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/overview/EventSinkServicesOverviewList.tsx]: yeah some sources may not have Pods or associated deployments so had this check

File: frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts [hunk 2]
```diff
@@ -10,26 +10,46 @@ const sinkServiceSchema = yup.object().shape({
...
+          .array()
+          .of(
+            yup.object({
+              apiVersion: yup.string().required('Required'),
+              kind: yup.string().required('Required'),
+            }),
+          )
+          .required('Required'),
+        mode: yup.string().max(253, 'Cannot be longer than 253 characters.'),
+        serviceAccountName: yup.string().max(253, 'Cannot be longer than 253 characters.'),
```
[the remark this answers is not in the document]
Comment (divyanshiGupta) [reply frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts]: Mode is also a dropdown field now. @invincibleJai I think the two validations can be removed.

[the remark this answers is not in the document]
Comment (invincibleJai) [reply frontend/packages/knative-plugin/src/components/add/eventSource-validation-utils.ts]: yes removed both :)

**Comment by sahil143** [comment]
/lgtm cancel

There need to be some changes in the Mode dropdown. @invincibleJai is looking into it

**Comment by invincibleJai** [comment]
> /lgtm cancel
> 
> There need to be some changes in the Mode dropdown. @invincibleJai is looking into it

Change was wrt mode value and it's `Ref` have updated , PTAL

https://github.com/knative/eventing/blob/6b9f75aaf425cbdef451ec069f0eab20371d0990/pkg/apis/sources/v1alpha1/apiserver_validation.go#L27

**Comment by divyanshiGupta** [comment]
Code looks good. Tested locally, was able to create ApiServerSource. 
/lgtm

**Comment by sahil143** [comment]
/lgtm

Mode dropdown changes works fine. Verified changes locally.

**Comment by serenamarie125** [comment]
General comment @invincibleJai - we will not have final text available for labels/help text at this time, so please don't worry about updating those now. We will have a follow up design story to work on that.

**Comment by invincibleJai** [comment]
@spadgett @benjaminapetersen needed your help with review of 0b76e21f69b8c27af4e6e2bc630613c11aee45b2

**Comment by invincibleJai** [comment]
@sahil143 have added `sticky` :)

**Pull request merged**